homeassistant/components/dweet.py

```diff
@@ -0,0 +1,68 @@
+"""
+A component which allows you to send data to Dweet.io.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/dweet/
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.const import EVENT_STATE_CHANGED, STATE_UNKNOWN
+from homeassistant.helpers import state as state_helper
+from homeassistant.helpers import validate_config
+from homeassistant.util import Throttle
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "dweet"
+DEPENDENCIES = []
+
+REQUIREMENTS = ['dweepy==0.2.0']
+
+CONF_WHITELIST = 'whitelist'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+
+
+# pylint: disable=too-many-locals
+def setup(hass, config):
+    """Setup the Dweet.io component."""
+    if not validate_config(config,
```

Comment:
We shouldn't use `validate_config` anymore. Use voluptuous instead and define a `CONFIG_SCHEMA`.